src/main/java/ca/on/oicr/gsi/dimsum/data/external/ExternalAssay.java

```diff
@@ -0,0 +1,35 @@
+package ca.on.oicr.gsi.dimsum.data.external;
+
+import ca.on.oicr.gsi.cardea.data.Assay;
+
+public class ExternalAssay {
```

Comment:
Would this be a good scenario for using Java `record` ? This class could be simplified to:
```
public record ExternalAssay  (long id, String name, String version, String description) {
  public ExternalAssay(Assay from) {
    this(from.getId(), from.getName(), from.getVersion(), from.getDescription());
  }
}
```
The constructors for the ExternalCase and some of the others would be a bit more involved, and would require some comments about e.g. `internal models x, y, z have no sensitive fields`, but the guarantees of immutability and built-in equals, hashCode and toString are nice.

Reply:
I thought the constructors were pretty gross when there were a lot of fields, but you're right - they're only used once inside the class, and the benefits easily outweigh that. Thanks for convincing me!